Security: Add clipboard button to 2FA time code for easier copying

abotic opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
## Fixes:
- https://github.com/Automattic/wp-calypso/issues/99827

## Proposed changes:
This PR adds a clipboard button next to the time code on the 2FA setup screen, allowing users to easily copy the code with a single click. This addresses user friction during the two-factor authentication setup process.

Key features:
* Adds a clipboard button next to the time code
* Provides clear visual feedback ("Copied!") when the code is copied
* Maintains existing design and component structure
* Improves usability for users setting up 2FA with an authenticator app

## Does this pull request change what data or activity we track or use?
Added a single analytics event to track when the 2FA time code is copied.

## Testing instructions:
1. Go to https://wordpress.com/me/security/two-step
2. Click on "Set up Two-Step Authentication"
3. On the setup screen, click on "Can't scan the code?" to view the time code
4. Verify you see the clipboard icon next to the time code
5. Click the clipboard icon and verify:
   - The code is copied to your clipboard
   - A "Copied!" message appears, confirming the action was successful
6. Paste the copied code somewhere to confirm it works correctly

## Example
https://github.com/user-attachments/assets/c572a5e5-fbaa-436e-9d79-509ed02038c3

abotic committed:
Security: Add clipboard button to 2FA time code for easier copying

katinthehatsite reviewed (commented):
In terms of functionality, everything worked for me 👍 

<img width="870" alt="Screenshot 2025-03-14 at 10 03 26 AM" src="https://github.com/user-attachments/assets/b7259bc8-ff6b-45cb-9a79-005075075767" />

One thing that I am not sure is the design. I see that we have different implementations of this, for example, in the celebrate modal:

<img width="631" alt="Screenshot 2025-03-14 at 10 12 12 AM" src="https://github.com/user-attachments/assets/170de6b9-d95c-458e-9fdf-0d66234703fd" />

For instance, the confirmation appearing in the tooltip vs. right by the button or using a `copy` icon vs `clipboard` icon (there are also other places where we use `copy` icon e.g. client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx ). I am okay to approve this because I think code-wise it makes sense but perhaps it might be worth to get some 👀 from a designer. What do you think?

abotic commented:
> In terms of functionality, everything worked for me 👍
> 
> <img alt="Screenshot 2025-03-14 at 10 03 26 AM" width="870" src="https://private-user-images.githubusercontent.com/25575134/422722705-b7259bc8-ff6b-45cb-9a79-005075075767.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE5NDcxMDgsIm5iZiI6MTc0MTk0NjgwOCwicGF0aCI6Ii8yNTU3NTEzNC80MjI3MjI3MDUtYjcyNTliYzgtZmY2Yi00NWNiLTlhNzktMDA1MDc1MDc1NzY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzE0VDEwMDY0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY2YWU0ODRiNTc3YmZkZjJlZmZiN2E5NmU5MWJiN2MzNWYwZDFjY2RhY2IwNmJiZWE5YmEzNjQ1NWY5NGI0YTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.FpeQCF8LB4Ostu0r2zg3Q5HuL82DWgm2NHa-YjnGxzQ">
> One thing that I am not sure is the design. I see that we have different implementations of this, for example, in the celebrate modal:
> 
> <img alt="Screenshot 2025-03-14 at 10 12 12 AM" width="631" src="https://private-user-images.githubusercontent.com/25575134/422723211-170de6b9-d95c-458e-9fdf-0d66234703fd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE5NDcxMDgsIm5iZiI6MTc0MTk0NjgwOCwicGF0aCI6Ii8yNTU3NTEzNC80MjI3MjMyMTEtMTcwZGU2YjktZDk1Yy00NThlLTlmZGYtMGQ2NjIzNDcwM2ZkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzE0VDEwMDY0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM1MjNlOGU3MmNjZmE4YWYyN2I3MzM3NzZlNTI5M2QzZmQ0ZDIxOTVlYmQwNzk2YWIzOGMyNWNhNzk5NzBjOGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.KPAFfGJzUfICBgwncJHv9vQ-_KhR0ZPg1dqz2iBfHJQ">
> For instance, the confirmation appearing in the tooltip vs. right by the button or using a `copy` icon vs `clipboard` icon (there are also other places where we use `copy` icon e.g. client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx ). I am okay to approve this because I think code-wise it makes sense but perhaps it might be worth to get some 👀 from a designer. What do you think?

Thanks for great feedback.

We do have several design styles for clipboard icon, the only reason I went with this one is that it was the closest one to match the font styles of code. However, I do agree we should check with design team for this decision.

I think this bug was reported as part of walkthrough where @matt-west and @jartes gave this insight.
Could you help us with decision here? Thanks

matt-west commented:
Thanks for the ping @abotic!

Let’s use the `copySmall` icon from @wordpress/icons here.

<img width="377" alt="Screenshot 2025-03-14 at 10 28 06" src="https://github.com/user-attachments/assets/4aa01d2d-8c74-4758-a316-6a04a2eb393c" />


https://wordpress.github.io/gutenberg/?path=/docs/icons-icon--docs

abotic committed:
Security: Update terminology from 'time code' to 'setup code' in 2FA setup (#101336)

* Update terminology from 'time code' to 'setup code' in 2FA setup

* fix for feedback suggestion

abotic committed:
Update clipboard icon and tooltip per comment

abotic committed:
Merge remote-tracking branch 'origin/trunk' into add/clipboard-to-2fa-setup-code

abotic commented:
> Thanks for the ping @abotic!
> 
> Let’s use the `copySmall` icon from @wordpress/icons here.
> 
> <img alt="Screenshot 2025-03-14 at 10 28 06" width="377" src="https://private-user-images.githubusercontent.com/417538/422746620-4aa01d2d-8c74-4758-a316-6a04a2eb393c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDIyMjU0NDUsIm5iZiI6MTc0MjIyNTE0NSwicGF0aCI6Ii80MTc1MzgvNDIyNzQ2NjIwLTRhYTAxZDJkLThjNzQtNDc1OC1hMzE2LTZhMDRhMmViMzkzYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMxN1QxNTI1NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05OGJjYTQwM2ZkNmFmMDIwY2NiNzY2ZGNlMTRiMTFkMDllZDJmNjNjMzVjYjI4ODVhMzVkNTY3NGVjZWNhOGJjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.8Q0w0elVZVDEjJUJKC1uRWwu84mEKO_juD6myI603y0">
> https://wordpress.github.io/gutenberg/?path=/docs/icons-icon--docs

Thanks for the suggestion @matt-west and @katinthehatsite !
I've implemented the copySmall icon and changed the way tooltip shows up to be more consistent with other parts of the site. New example video in description is replaced so you can see how it looks now.

Let me know know if you think some other changes are needed.

matt-west commented:
Looks good to me now. Thanks @abotic

rcrdortiz reviewed (approved):
Changes look good to me and all feedback has been addressed.

abotic committed:
Merge remote-tracking branch 'origin/trunk' into add/clipboard-to-2fa-setup-code
